src/transaction.js

```diff
@@ -130,18 +131,25 @@ export class Transaction {
       throw new Error('Invalid publicKey');
     }
 
+    let keypair;
     let hint;
+    const signatureBuffer = Buffer.from(signature, 'base64');
 
     try {
-      hint = Keypair.fromPublicKey(publicKey).signatureHint();
+      keypair = Keypair.fromPublicKey(publicKey);
+      hint = keypair.signatureHint();
     } catch (e) {
       throw new Error('Invalid publicKey');
     }
 
+    if (!keypair.verify(this.hash(), signatureBuffer)) {
```

Comment:
This isn't a safe assumption to make. This only validates the "master key", but doesn't account for other signers that may be on the account. See the [docs on multisignature](https://www.stellar.org/developers/guides/concepts/multi-sig.html#additional-signing-keys) for more info.

Reply:
I am not sure I understand your comment. This verifies that the provided `signature` is a valid signature of the keypair provided by the `publicKey` for this transaction.

So this does not verify the master key but it verifies the signer that provided the signature – as intended. Am I missing something?